### ISSUE TITLE
False prediction when head is oriented

### ISSUE THREAD

**Issue by abdou31** [TadasBaltrusaitis/OpenFace#719]
Hello Tadas ,
I have writed a issue that explain what i want to do ( annotating dataset using OpenFace ) to train this dataset by CNN Facial landmarks , i have get false prediction , after verify the dataset and points , i have found some problems that due to this false results.
this is a annotation created by OpenFace : 
https://user-images.githubusercontent.com/19480228/59348270-f58d8400-8d16-11e9-9949-70f24c292b93.jpg

Do you have any idea how can i correct this prediction when the head is oriented ?
Note : I don't use 3D values ( coordinates points ) .
Thanks

**Comment by TadasBaltrusaitis** [comment]
It is a reasonably challenging problem knowing if landmark detection succeeded or failed (if it was easy it would be simple to build a perfect landmark detector). 

One thing you could possibly use is the likelihood parameters in OpenFace that can estimate how reliable the landmark detection is (each landmark has a likelihood term). However, these are not included in the csv file and you would have to modify the code in order to access and output them.

You can access them through `face_model.landmark_likelihoods`

**Comment by abdou31** [comment]
You mean that i should download the project source not the binary (exe project ), modify the code and build again openface project 
If yes , can you tell me what should i change exactly? 
I mean what file should i modifying

**Comment by TadasBaltrusaitis** [comment]
Exactly, you will need to change the source code yourself. The changes will not be trivial, so I cannot give you exact instructions, but some rough guidelines:
1. You will want to access the `face_model.landmark_likelihoods` for both the 68 landmark points of the face and 28 + 28 landmark points of the eye region
2. You will want to output their values as part of the RecorderCSV to be able to output them as part of the CSV file
3. You will need to work out what is a good cutoff for trusting the landmark detection of them.

**Comment by abdou31** [comment]
Ok thanks Tadas , I will see that
